pkg/dockerregistry/server/util.go

```diff
@@ -21,9 +22,12 @@ import (
 )
 
 func getNamespaceName(resourceName string) (string, string, error) {
-	repoParts := strings.SplitN(resourceName, "/", 2)
+	repoParts := strings.Split(resourceName, "/")
 	if len(repoParts) != 2 {
-		return "", "", ErrNamespaceRequired
+		return "", "", distribution.ErrRepositoryNameInvalid{
+			Name:   resourceName,
+			Reason: fmt.Errorf("invalid repository name %q: it must be of the format <project>/<name>", resourceName),
+		}
```

Comment:
The resulting error will be weird: `repository name %q invalid: invalid repository name %q: it must be of the format <project>/<name>`